server/context.go

```diff
@@ -122,22 +134,18 @@ func (s *SessionManager) NewContext(conn *mysql.Conn) (*sql.Context, error) {
 func (s *SessionManager) getOrCreateSession(ctx context.Context, conn *mysql.Conn) (sql.Session, *sql.IndexRegistry, *sql.ViewRegistry, error) {
 	s.mu.Lock()
 	sess, ok := s.sessions[conn.ConnectionID]
-	ir := s.idxRegs[conn.ConnectionID]
-	vr := s.viewRegs[conn.ConnectionID]
-	if !ok {
-		var err error
-		sess, ir, vr, err = s.builder(ctx, conn, s.addr)
+	s.mu.Unlock()
```

Comment:
Not an actionable comment, but just something I wish Go had. In C# they have the `lock` command where some block of code executes under a mutex, guaranteeing that the mutex is always released once leaving the block's scope. In Go we have to declare a function and use `defer` to get something equivalent, while adding some pains with return handling and whatnot. Just wishing...

Reply:
Yeah go locking is lame, I blame Rob Pike